pallets/ocex/src/lib.rs

```diff
@@ -211,11 +215,33 @@ pub mod pallet {
 					));
 				});
 				<Accounts<T>>::insert(&main_account, account_info);
-				Self::deposit_event(Event::MainAccountRegistered { main: main_account, proxy });
+				Self::deposit_event(Event::NewProxyAdded { main: main_account, proxy });
 			}
 			Ok(())
 		}
 
+		/// Removes a proxy account from pre-registered main acocunt
+		#[pallet::weight(10000)]
+		pub fn remove_proxy_account(origin: OriginFor<T>, proxy: T::AccountId) -> DispatchResult {
+			let main_account = ensure_signed(origin)?;
+			ensure!(<Accounts<T>>::contains_key(&main_account), Error::<T>::MainAccountNotFound);
+			<Accounts<T>>::try_mutate(&main_account, |account_info| {
+				if let Some(account_info) = account_info {
+					let proxy_positon = account_info.proxies.iter().position(|account| *account == proxy).ok_or(Error::<T>::ProxyNotFound)?;
+					ensure!(account_info.proxies.len() > 1, Error::<T>::MinimumOneProxyRequired);
```

Comment:
If minimum 1 should be present then I assume condition check should be related to 0, e.g.:
```rust
ensure!(account_info.proxies.len() > 0, Error::<T>::MinimumOneProxyRequired);
```
And in overall - should we return/specify this? we can just return `ProxyNotFound` if there are no proxies...

Reply:
It is to stop you from removing your only proxy, it should be account_info.proxies.len() > 1 as you need to have at least 2 to be able to remove one, and needs it own error to explain that condition.

Reply:
Ok, thanks.

Reply:
It should be `account_info.proxies.len() > 1`, Also, Orderbook in enclave should be notified of such change other wise orders placed from this proxy account will be inaccessible, @zktony you can use IngressMessages storage to let enclave know about this change and react accordingly.